File: JANUARY/03-01-2026/1411.rb
# @param {Integer} n
# @return {Integer}
def num_of_ways(n)
    mod = 1_000_000_007
    aba = 6
    abc = 6
    (2..n).each do
        new_aba = (aba * 3 + abc * 2) % mod
        new_abc = (aba * 2 + abc * 2) % mod
        aba = new_aba
        abc = new_abc
    end
    (aba + abc) % mod
end

File: JANUARY/03-01-2026/1411.java
class Solution {
    public int numOfWays(int n) {
        final int MOD = 1000000007;
        long aba = 6, abc = 6;
        for (int i = 2; i <= n; i++) {
            long newAba = (aba * 3 + abc * 2) % MOD;
            long newAbc = (aba * 2 + abc * 2) % MOD;
            aba = newAba;
            abc = newAbc;
        }
        return (int)((aba + abc) % MOD);
    }
}